diff --git a/scripts/codegen/generate-artifacts-executor/generateRCTThirdPartyComponents.js b/scripts/codegen/generate-artifacts-executor/generateRCTThirdPartyComponents.js
--- a/scripts/codegen/generate-artifacts-executor/generateRCTThirdPartyComponents.js
+++ b/scripts/codegen/generate-artifacts-executor/generateRCTThirdPartyComponents.js
@@ -138,7 +138,7 @@
   const thirdPartyComponentsMapping = Object.keys(componentsInLibraries)
     .flatMap(library => {
       const components = componentsInLibraries[library];
       return components.map(({componentName, className}) => {
-        return `\t\t@"${componentName}": NSClassFromString(@"${className}"), // ${library}`;
+        return `\t\t{ Class _cls = NSClassFromString(@"${className}"); if (_cls) { components[@"${componentName}"] = _cls; } } // ${library}`;
       });
     })
     .join('\n');
diff --git a/scripts/codegen/templates/RCTThirdPartyComponentsProviderMM.template b/scripts/codegen/templates/RCTThirdPartyComponentsProviderMM.template
--- a/scripts/codegen/templates/RCTThirdPartyComponentsProviderMM.template
+++ b/scripts/codegen/templates/RCTThirdPartyComponentsProviderMM.template
@@ -21,8 +21,8 @@
 
   dispatch_once(&nativeComponentsToken, ^{
-    thirdPartyComponents = @{
+    NSMutableDictionary<NSString *, Class<RCTComponentViewProtocol>> *components = [NSMutableDictionary new];
 {thirdPartyComponentsMapping}
-    };
+    thirdPartyComponents = [components copy];
   });
 
   return thirdPartyComponents;
